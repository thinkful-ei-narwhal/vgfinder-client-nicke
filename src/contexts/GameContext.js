import React, { Component } from "react";

export const nullGame = {
  id: "",
  title: "",
  description: "",
  genre: "",
  rating: "",
  release_date: "",
  developer: "",
  trailer_url: "",
  image_url_box_art: "",
  image_url_two: "",
  image_url_three: "",
  image_url_four: "",
  image_url_five: "",
};

const GameContext = React.createContext({
  game: nullGame,
  setGame: () => {},
  clearGame: () => {},
});

export default GameContext;

export class GameProvider extends Component {
  state = {
    game: nullGame,
  };

  setGame = (game) => {
    this.setState({ game });
  };

  clearGame = () => {
    this.setGame(nullGame);
  };

  render() {
    const value = {
      game: this.state.game,
      setGame: this.setGame,
      clearGame: this.clearGame,
    };
    return (
      <GameContext.Provider value={value}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}
