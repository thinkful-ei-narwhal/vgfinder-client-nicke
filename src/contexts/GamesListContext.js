import React, { Component } from 'react';

const GamesListContext = React.createContext({
  gamesList: [],
  error: null,
  setError: () => { },
  clearError: () => { },
  setGamesList: () => { },
})
export default GamesListContext

export class GamesListProvider extends Component {
  state = {
    gamesList: [],
    error: null,
  };

  setGamesList = gamesList => {
    this.setState({ gamesList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      gamesList: this.state.gamesList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setGamesList: this.setGamesList,
    }
    return (
      <GamesListContext.Provider value={value}>
        {this.props.children}
      </GamesListContext.Provider>
    )
  }
}
