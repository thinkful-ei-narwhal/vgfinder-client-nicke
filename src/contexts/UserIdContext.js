import React, { Component } from 'react'

const UserIdContext = React.createContext({
  userId: undefined,
  wishlist: [],
  setUserId: () => { },
  setUserWishlist: () => { },
})

export default UserIdContext

export class UserIdContextProvider extends Component {
  state = {
    userId: undefined,
    wishlist: []
  };

  setUserId = (userId) => {
    this.setState({ userId })
  }

  setUserWishlist = (wishlist) => {
    this.setState({ wishlist })
  }


  render() {
    const value = {
      userId: this.state.userId,
      wishlist: this.state.wishlist,
      setUserId: this.setUserId,
      setUserWishlist: this.setUserWishlist
    }

    return (
      <UserIdContext.Provider value={value}>
        {this.props.children}
      </UserIdContext.Provider>
    )
  }
}
