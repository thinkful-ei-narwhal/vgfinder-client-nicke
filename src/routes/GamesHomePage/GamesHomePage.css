.GamesListPage {
  margin-bottom: 30px;
}
