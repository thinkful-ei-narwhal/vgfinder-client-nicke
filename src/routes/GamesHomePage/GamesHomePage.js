import React, { Component } from 'react';
import GamesApiService from '../../services/games-api-service';
import GamesListContext from '../../contexts/GamesListContext';
import { Section } from '../../components/Utils/Utils';
import Header from './../../components/Header/Header';
import './GamesHomePage.css'
import GameInfo from '../../components/GameInfo/GameInfo';
import GamesCarousel from '../../components/GameCarousel/GameCarousel';

export default class GamesHomePage extends Component {
  static contextType = GamesListContext;

  constructor(props) {
    super(props)
    this.state = {
      loaded: false,
      activeGameId: 0
    }
  }

  setActiveGame = (activeGameId) => {
    this.setState({ activeGameId });
  }

  componentDidMount() {
    this.context.clearError()
    GamesApiService.getGames()
      .then(res => {
        this.context.setGamesList(res)
        this.setState({ loaded: true, activeGameId: res[0].id })
      })
      .catch(this.context.setError)
  }

  renderGamesHomePage() {
    const { error } = this.context
    const games = this.context.gamesList.map(game => game);
    const reel = games.map(game => { return { imgUrl: game.image_url_box_art, gameId: game.id } });
    const activeGame = games.find(game => game.id === this.state.activeGameId);
    return <>
      <Header pathName={this.props.location.pathname} />
      <Section className='GamesListPage'>
        <h2>{activeGame.title}</h2>
        {error ? <p className='red'>There was an error, try again</p>
         : <GamesCarousel reel={reel} isSingleGame={false} setActiveGame={this.setActiveGame} />}
        <GameInfo game={activeGame} />
      </Section>
    </>
  }

  render() {
    return (
      <>
        {this.state.loaded ? this.renderGamesHomePage() : null}
      </>
    )
  }
}