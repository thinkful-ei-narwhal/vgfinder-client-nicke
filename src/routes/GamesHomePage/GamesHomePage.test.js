import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GamesHomePage from "./GamesHomePage";
import { GamesListProvider } from "./../../contexts/GamesListContext";

describe(`GamesHomePage component`, () => {
  it("renders a GamesHomePage component", () => {
    const wrapper = shallow(
      <GamesListProvider>
        <GamesHomePage />
      </GamesListProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
