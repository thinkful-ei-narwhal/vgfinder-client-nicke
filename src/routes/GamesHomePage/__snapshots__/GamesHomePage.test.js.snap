// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesHomePage component renders a GamesHomePage component 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "error": null,
      "gamesList": Array [],
      "setError": [Function],
      "setGamesList": [Function],
    }
  }
>
  <GamesHomePage />
</ContextProvider>
`;
